quipucords/api/source/serializer.py

```diff
@@ -273,228 +276,130 @@ def validate_name(name):
         return name
 
     @staticmethod
-    def validate_ipaddr_list(hosts):  # noqa: PLR0912, PLR0915, C901
-        """Make sure the hosts list is present and has valid IP addresses."""
-        ipaddr_list = hosts
-        if isinstance(ipaddr_list, list):
-            ipaddr_list = [item for item in ipaddr_list if item]
+    def validate_ipaddr_list(hosts):
```

Comment:
This is maybe a minor nitpick, but to make this more readable, I would rename this function to `validate_addresses` because the current title implies that only IP addresses are allowed or expected, and the `_list` suffix is redundant with a plural name. If you want to be extra clear to the reader, we could add type hints like:

```suggestion
    def validate_addresses(addresses: list) -> list:
```

…and I would rename `hosts` (and related variables like `normalized_hosts`) to something like `addresses` (and `normalized_addresses`) since the original name may also be slightly confusing or _contradict_ the original function name. Thinking as a naive reader might: "Are these IP addresses? Or hostnames? Or actual hosts (systems)? Or both or neither?"

Reply:
I do realize that a lot of legacy code here and outside here uses the word "host", which I'm not terribly fond of, and I also realize there is some subjective judgment here. Part of my thought process is that we should clean up the language while we are in the neighborhood, especially if renaming things might make them more immediately comprehensible for future readers (or future us), because we likely will not touch the code in this area for a long time after this PR completes.